src/osp/Resource/AssetImporter.cpp

```diff
@@ -158,8 +145,26 @@ void osp::AssetImporter::load_part(TinyGltfImporter& gltfImporter,
             }
         }
 
-        part.get_machines().emplace_back(std::move(machine));
+        machineIndexArray.push_back(machineArray.size());
+        machineArray.emplace_back(std::move(machine));
```

Comment:
PrototypeMachines are added to the PrototypePart's machine array, and the indices are stored in the PrototypeObject's machineIndexArray

Reply:
Make sure to reserve space in the vector before calling emplace in a loop.

Note though that if this function is being called in a loop, don't reserve a whole bunch, instead figure out a way to get the total size the vect needs to be, and reserve once at the top of the stack.

Reply:
For the `machineIndexArray`, that's easy (and I may actually just change that parameter to a return value). For the `machineArray`, it's not so easy. There's not really any way to know how many machines will be in the part's hierarchy unless we add a second recursion pass just to pre-allocate machine storage. I think it's probably best to just let `std::vector` do its thing and then maybe shrink it to fit afterwards, unless you think the allocations are really such a big deal that it's worth doing a pre-pass to figure out how much space it needs. Otherwise, I'll just leave a comment on line 149 for now that pre-allocation would be nice, and if we ever find ourselves needing to do multiple passes over the config for some reason, we can add this at that point.